portfolio/src/main/java/com/google/sps/data/FindSchedule.java

```diff
@@ -0,0 +1,116 @@
+      Comparator.comparing(Task::getDurationSeconds).thenComparing(Task::getName);
+
+  /**
+   * This method returns a Task Collection which the greedy algorithm
+   * determines should be scheduled for the person.
+   */
+  public Collection<Task> greedy(Collection<CalendarEvent> events, Collection<Task> tasks,
+      long workHoursStartTimeSeconds, long workHoursEndTimeSeconds) {
+    List<CalendarEvent> eventsList = new ArrayList<CalendarEvent>(events);
+    Task[] tasksArray = tasks.toArray(new Task[tasks.size()]);
+    Collections.sort(eventsList, sortAscending);
+    Arrays.sort(tasksArray, 0, tasks.size(), sortByDuration);
+    TimeRange[] availableTimes =
+        emptyTimeRanges(eventsList, workHoursStartTimeSeconds, workHoursEndTimeSeconds);
+
+    List<Task> scheduledTasks = new ArrayList<Task>();
+    // Index of TimeRange we are in
+    int i = 0;
+    // Index of Task we are trying to schedule
+    int j = 0;
+    // Integer indicating the start time (in seconds) we are currently trying to
+    // schedule events in
+    long scheduleTimeSeconds = 0;
+
+    // This will iterate through the time ranges and tasks and if one can be
+    // scheduled then it will be (and we move onto the next task) otherwise
+    // we move on to the next range (this is because the tasks are sorted by
+    // duration so if one task did not fit in the given range then we know no
+    // later ones will fit either). We create new Task objects for the result
+    // so data structures passed in are never changed.
+    while (i < availableTimes.length && j < tasksArray.length) {
+      TimeRange timeRange = availableTimes[i];
+      Task task = tasksArray[j];
+      // Either time is already past the start of the time range or we should
+      // update it (maybe this is our first iteration in the range).
+      scheduleTimeSeconds = Math.max(scheduleTimeSeconds, timeRange.start());
+      // The task can be scheduled in the current time range.
+      if (scheduleTimeSeconds + task.getDurationSeconds() <= timeRange.end()) {
+        Task scheduledTask;
+        if (task.getDescription().isPresent()) {
+          scheduledTask =
+              new Task(task.getName(), task.getDescription().get(), task.getDuration().toMinutes(),
+                  task.getPriority(), Instant.ofEpochSecond(scheduleTimeSeconds).toString());
+        } else {
+          scheduledTask = new Task(task.getName(), null, task.getDuration().toMinutes(),
+              task.getPriority(), Instant.ofEpochSecond(scheduleTimeSeconds).toString());
+        }
+
+        scheduledTasks.add(scheduledTask);
+        scheduleTimeSeconds += task.getDurationSeconds();
+        j++;
+      } else {
+        i++;
+      }
+    }
+    return scheduledTasks;
+  }
+
+  /**
+   * This method returns an TimeRange array which represent the periods of
+   * time that are empty of events and lie completely inside the person's
+   * working hours.
+   */
+  public static TimeRange[] emptyTimeRanges(
```

Comment:
1. Is there a reason why you return array instead of list? Lists are more flexible and easier to work with. Consider returning a list of TimeRange.
2. I think it'd be nice to clarify in the method name that it gets or computes the time ranges. So for example, computeEmptyTimeRanges or getEmptyTimeRanges.

Reply:
Changed it to return a List instead and renamed it to getEmptyTimeRanges.